lib/pages/guide.js

```diff
@@ -79,13 +70,47 @@ function reindexFootnoteListing(footnotesDiv, footnoteMap) {
 }
 
 function reindexFootnotes(bodyHast) {
-  const footnotesDiv = bodyHast.children.filter(child => child.tagName === 'div' && child.properties.className.includes('footnotes'))[0];
-  if(!footnotesDiv) { throw new Error(`No footnotes div found. Did you call unified.use(footnotes)?`)};
+  const footnotesDiv = bodyHast.children.filter(child => child.tagName === 'div' && child.properties.className && child.properties.className.includes('footnotes'))[0];
+  if(!footnotesDiv) {
+    throw new Error(`No footnotes div found. Did you call unified.use(footnotes)?`);
+  };
 
   const footnoteMap = reindexFootnoteMarkersUnderNode(bodyHast);
   reindexFootnoteListing(footnotesDiv, footnoteMap);
 }
 
+function htmlDirectives() {
+  return (tree) => {
+    visit(tree, 'image', configureImage);
+    visit(tree, 'blockquote', configureBlockquotes);
+  };
+
+  function configureImage(node) {
+    const hast = h('image', {
+      ...node,
+      className: "inline",
+      align: 'right',
+    });
+    
+    node.data = {
+      hName: hast.tagName,
+      hProperties: hast.properties,
+    };
+  }
+  
+  function configureBlockquotes(node) {
+    const headerText = node.children
+      .filter(node => node.type === 'heading')[0].children
+      .filter(node => node.type === 'text')[0].value;
+        
+    node.data = {
+      hProperties: {
+        className: SECTION_CLASSES[headerText],
+      },
+    };
```

Comment:
This is a nice way to fold that previous function into this paradigm! I think I am still in favor of trying to do this via a directive in the markdown (`:attrs{.did-you-know}` or something maybe?) but that's an enhancement for later, not a necessity for now.

Reply:
I can't seem to make directives work; they just don't show up ever

Reply:
we'll see if I get around to that before pushing this